src/error.rs

```diff
@@ -1,18 +1,63 @@
-use thiserror::Error;
+use std::{
+    error::Error as StdError,
+    fmt::{Display, Formatter, Result as FmtResult},
+};
 
-pub type YoutubeResult<T> = Result<T, YoutubeError>;
+use reqwest::Error as ReqwestError;
 
-#[derive(Debug, Error)]
-pub enum YoutubeError {
-    #[error(transparent)]
-    Reqwest(#[from] reqwest::Error),
+pub type Result<T> = std::result::Result<T, Youtube>;
 
-    #[error("Parse Error: {0}")]
-    ParseError(String),
+#[derive(Debug)]
+pub enum Youtube {
```

Comment:
I don't like the renaming. The purpose of the enum should be explicitly stated in the name. Calling it `Error` would be more fitting.